lib/txRequest/txRequest.js

```diff
@@ -93,10 +95,14 @@ class TxRequest {
     }
 
     if (this.temporalUnit == 1) {
-      return new BigNumber(await Util.getBlockNumber(this.web3))
+      // The reason for the `plus(1)` here is that the next block to be mined
+      // is for all intents and purposes the `now` since the soonest this transaction
+      // could be included is alongside it.
+      return new BigNumber(await Util.getBlockNumber(this.web3)).plus(1);
     } else if (this.temporalUnit == 2) {
-      const timestamp = await Util.getTimestamp(this.web3)
-      return new BigNumber(timestamp)
+      const timestamp = new BigNumber(await Util.getTimestamp(this.web3));
+      const local = new BigNumber(moment().seconds());
+      return local.gt(timestamp) ? local : timestamp;
```

Comment:
Not sure if that `if` is needed. We need to assume that TimeNode has synced clock anyways.

```suggestion
      return local;
```